HW2/P2/parallel_vector.pyx

```diff
@@ -107,9 +113,20 @@ cpdef move_data_medium_grained(np.int32_t[:] counts,
    ##########
    with nogil:
        for r in range(repeat):
-           for idx in range(src.shape[0]):
+           for idx in prange(src.shape[0], num_threads=4):
                if counts[src[idx]] > 0:
-                   counts[dest[idx]] += 1
-                   counts[src[idx]] -= 1
+                   if src[idx]/N == dest[idx]/N:
```

Comment:
@isadoranun you are right, I oversaw that. Nevertheless, I would recommend to improve this code strongly by using a new temporary variable 

```
sidx = src[idx] / N # floored

#or the actual meaning that is used
 
sidx = <int>floor(src[idx] / N)
```
the reason for this is that it is quite difficult to get the meaning out of this code (noone looks up type definitions that are many lines away):

(1) depending on the type the result will be different. => python is not typesafe, so an actual floor operation should be not obfuscated by assuming types, even in cython code. For C/C++ code or any other typesafe language this is ok, but for languages relying on dynamic typing this is an error source.

I.e. try (which is not good code at all as it has many flaws)
```
arr = [1, 2, 3, 4, 5]

# as division is slow, a common optimization includes multiplying by the inverse!
m = 0.5 * len(arr)

arr1 = arr[:m]
arr2 = arr[m:]
```

(2) the operation is used multiple times => store it in a temporary variable, which makes the code better readable and also helps the compiler